### PR TITLE
Add API key validation for AI models

### PR DESCRIPTION
**Pull request by Gyan172004**
## Proposed Changes

- Added validation for required API keys (gpt4-o, gemini-pro, mistral, claude)

- Enhanced error messaging to guide users when API keys are missing

### CODE REVIEW

File: src/popup/popup.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ import {
   loadProfiles,
 } from '@utils/storage/profiles/profileManager';
 
+import { validateLLMConfiguration } from '../utils/missingApiKey';
```
Comment (rootCircle) [inline src/popup/popup.ts]: Use @utils/missingApiKey syntax for better clarity!

File: src/utils/storage/getProperties.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import LLMEngineType from './llmEngineTypes';
+import LLMEngineType from '../llmEngineTypes';
```
Comment (rootCircle) [inline src/utils/storage/getProperties.ts]: Use @utils/... like syntax here as well

**Review by rootCircle — CHANGES_REQUESTED** [review]

File: src/utils/storage/getProperties.ts (hunk 2 above)
Comment (Gyan172004) [reply src/utils/storage/getProperties.ts]: Done

File: src/popup/popup.ts (hunk 1 above)
Comment (Gyan172004) [reply src/popup/popup.ts]: Done

**Review by rootCircle — APPROVED** [review]
Super thanks for this, highly needed!

**Pull request merged**